packages/react/src/Button/Button.test.tsx

```diff
@@ -104,10 +105,22 @@ describe('Button', () => {
   it('is able to pass a React ref', () => {
     const ref = createRef<HTMLButtonElement>()
 
-    const { container } = render(<Button ref={ref} />)
+    const { container } = render(<Button ref={ref}>Click me!</Button>)
 
     const button = container.querySelector(':only-child')
 
     expect(ref.current).toBe(button)
   })
+
+  it('renders a button with an icon', () => {
```

Comment:
Can we test the two `iconPosition` values as well? E.g. `start` should actually render it as the first child.

Reply:
```html
<button>
 Click me!
 <icon />
</button>
```

In this scenario `.ams-button > .ams-icon:first-child` would apply to the icon too. As well as `last-child`.

Reply:
Oh yeah. Bummer.

Reply:
Fixed the `undefined` and created tests for all icon positions using a `span` for the child.